tests/integration/grpc_stub_server.go

```diff
@@ -0,0 +1,87 @@
+package integration
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"google.golang.org/grpc"
+	testpb "google.golang.org/grpc/test/grpc_testing"
+)
+
+// StubServer is borrowed from the interal package of grpc-go.
+// See https://github.com/grpc/grpc-go/blob/master/internal/stubserver/stubserver.go
+// Since it cannot be imported directly, we have to copy and paste it here,
+// and useless code for our testing is removed.
+
+// StubServer is a server that is easy to customize within individual test
+// cases.
+type StubServer struct {
```

Comment:
Nice.

I would consider putting this into: 
     `etcd/pkg/testutil`
or  even  `etcd/pkg/grpc_testing`(slightly preferred)

as this is etcd-independent and fulfills criteria for pkg.

Reply:
Done